matrix-sdk-android/src/test/java/org/matrix/android/sdk/internal/session/room/aggregation/livelocation/LiveLocationAggregationProcessorTest.kt

```diff
@@ -0,0 +1,387 @@
+        val beaconInfoNotLive = MessageBeaconInfoContent(isLive = false)
+
+        val resultLiveNoEventId = liveLocationAggregationProcessor.handleBeaconInfo(
+                realm = fakeRealm.instance,
+                event = eventWithNoEventId,
+                content = beaconInfoLive,
+                roomId = A_ROOM_ID,
+                isLocalEcho = false
+        )
+        val resultLiveEmptyEventId = liveLocationAggregationProcessor.handleBeaconInfo(
+                realm = fakeRealm.instance,
+                event = eventWithEmptyEventId,
+                content = beaconInfoLive,
+                roomId = A_ROOM_ID,
+                isLocalEcho = false
+        )
+        val resultNotLiveNoEventId = liveLocationAggregationProcessor.handleBeaconInfo(
+                realm = fakeRealm.instance,
+                event = eventWithNoEventId,
+                content = beaconInfoNotLive,
+                roomId = A_ROOM_ID,
+                isLocalEcho = false
+        )
+        val resultNotLiveEmptyEventId = liveLocationAggregationProcessor.handleBeaconInfo(
+                realm = fakeRealm.instance,
+                event = eventWithEmptyEventId,
+                content = beaconInfoNotLive,
+                roomId = A_ROOM_ID,
+                isLocalEcho = false
+        )
+
+        resultLiveNoEventId shouldBeEqualTo false
+        resultLiveEmptyEventId shouldBeEqualTo false
+        resultNotLiveNoEventId shouldBeEqualTo false
+        resultNotLiveEmptyEventId shouldBeEqualTo false
+    }
+
+    @Test
+    fun `given beacon info and existing entity when beacon content is correct and active then it is aggregated`() {
+        val event = Event(
+                senderId = A_SENDER_ID,
+                eventId = AN_EVENT_ID
+        )
+        val beaconInfo = MessageBeaconInfoContent(
+                isLive = true,
+                unstableTimestampMillis = A_TIMESTAMP,
+                timeout = A_TIMEOUT_MILLIS
+        )
+        fakeClock.givenEpoch(A_TIMESTAMP + 5000)
+        val aggregatedEntity = mockLiveLocationShareAggregatedSummaryEntityForEvent()
+        val previousEntities = mockPreviousLiveLocationShareAggregatedSummaryEntities()
+
+        val result = liveLocationAggregationProcessor.handleBeaconInfo(
+                realm = fakeRealm.instance,
+                event = event,
+                content = beaconInfo,
+                roomId = A_ROOM_ID,
+                isLocalEcho = false
+        )
+
+        result shouldBeEqualTo true
+        aggregatedEntity.eventId shouldBeEqualTo AN_EVENT_ID
+        aggregatedEntity.roomId shouldBeEqualTo A_ROOM_ID
+        aggregatedEntity.userId shouldBeEqualTo A_SENDER_ID
+        aggregatedEntity.isActive shouldBeEqualTo true
+        aggregatedEntity.endOfLiveTimestampMillis shouldBeEqualTo A_TIMESTAMP + A_TIMEOUT_MILLIS
+        aggregatedEntity.lastLocationContent shouldBeEqualTo null
+        previousEntities.forEach { entity ->
```

Comment:
``kotlin`
val previousEntities = mockPreviousLiveLocationShareAggregatedSummaryEntities()

previousEntities.forEach { entity ->
  entity.isActive shouldBeEqualTo false
}
```

it looks like the `previousEntities` aren't used?

Reply:
In fact, they are. They are modified internally by the aggregation process. And here we check the result is correct. I could not find a better way to ensure the aggregation is correct. I am open to any suggestion that could improve that.